XIVComboPlugin/IconReplacer.cs

```diff
@@ -635,6 +635,14 @@ private ulong GetIconDetour(byte self, uint actionID)
                         return SMN.EnergySyphon;
                     return SMN.Painflare;
                 }
+            
+            //Change Summon Solar Bahamut into Lux Solaris
+            if(Configuration.ComboPresets.HasFlag(CustomComboPreset.SummonerSolarBahamutLuxSolaris))
+                if ( iconHook.Original(self, actionID) == SMN.SummonSolarBahamut )
```

Comment:
Note that Summon Bahamut is the only demi-summon ability that can be put on your hotbar.
```suggestion
                if (actionID == SMN.SummonBahamut)
```

Reply:
Using iconHook.Original == SummonSolarBahamut was the only permutation I could find that would actually swap correctly. I just tried actionID == SMN.SummonBahamut locally and the swapping functionality appears to be broken.

Reply:
1. Don't forget `return iconHook.Original(self, actionID);`. I know it's at the bottom but still.
2. Out of curiosity, did you drag Summon Bahamut from your spellbook to your hotbar recently?

Reply:
Ah, I think my icon was dropped onto my hotbars many expansions ago, but I will readily admit that I am new enough to FF14 addon development that I don't fully understand the intricacies of the icon layers yet 😁

It looks like returning the original icon paired with re-dropping my bahamut summon onto my bar appears to work now, so thank you for the feedback! Could you explain why we prefer not to use the original icon unless its for a return call? You mentioned it on the VPR PR as well. My gut estimate is that the call could be expensive and so we only want to call it when we absolutely need to rather than for every pass of the IF check, but correct me if I'm wrong.

Thanks!

Reply:
I was in the middle of typing this up and I went to check some Action IDs for a complete example, and then I discovered they've changed the game's behavior a bit.

Hotbars store the Action ID of whatever was dragged from your spellbook. The game then runs the function this plugin hooks over every hotbar slot, to determine the "real" ability to show. So if you drag Jolt 1 onto your hotbar, the game will store 7524 in that slot. If you drag Jolt 2 there, in the past it would store a DIFFERENT Action ID. *HOWEVER*, it seems that with DT, all abilities that are flat upgrades have the same Action ID. Regardless, what is getting passed into this function is the hotbar-level Action ID, which may or may not match the Action that's actually displayed on the hotbar.

Reply:
So in your case, what likely happened was that you had an old Action ID for Summon Bahamut on your hotbar, which has been "migrated" by the client via the vanilla version of this function to give you the actual useful ability.

Reply:
Ah okay, thanks for the explanation. That makes sense to me and I could definitely see my cached action ID causing issues since I think it was placed there back in the 70-80 tier, especially if there's been changes recently. 

Is my intuition around the iconhook.Original call correct as well, in that the reason why we try to avoid calling it is due to expensive processing time?